Content.Server/VendingMachines/VendingMachineComponent.cs

```diff
@@ -21,13 +22,18 @@
 using Robust.Shared.Random;
 using Robust.Shared.Serialization.Manager.Attributes;
 using Robust.Shared.ViewVariables;
+using Robust.Shared.Containers;
+using Content.Shared.Whitelist;
+using Content.Server.Hands.Components;
+using Content.Server.Items;
+using Content.Shared.Popups;
 using static Content.Shared.Wires.SharedWiresComponent;
 
 namespace Content.Server.VendingMachines
 {
     [RegisterComponent]
     [ComponentReference(typeof(IActivate))]
-    public class VendingMachineComponent : SharedVendingMachineComponent, IActivate, IBreakAct, IWires
+    public class VendingMachineComponent : SharedVendingMachineComponent, IActivate, IBreakAct, IWires, IInteractUsing
```

Comment:
No new code should be using the obsolete interfaces. IMO this should also be a separate component.